### ISSUE TITLE
GHC-8.8.1 / base-4.13 / MonadFail / ... breakages

### ISSUE THREAD

**Issue by phadej** [haskell-infra/hackage-trustees#222]
- [x] aeson
- [x] basement
- [x] binary (0.8 ... 0.8.6)
- [x] cereal
- [x] colour
- [x] control-monad-omega
- [x] dlist
- [x] free
- [x] json
- [x] jose
- [x] language-c
- [x] machines
- [x] optparse-applicative
- [x] pipes
- [x] prettyprinter
- [x] socks
- [x] QuickCheck
- [x] text (1.2.3.0, 1.2.3.1)
- [x] time
- [x] transformers-0.5.5.2
- [x] vector-th-unbox-0.2.1.6

See also
- https://github.com/hvr/head.hackage/pull/111
- https://github.com/hvr/head.hackage/pull/112
- https://github.com/hvr/head.hackage/pull/117
- some packages have bounds in place, some doesn't.

**Comment by RyanGlScott** [comment]
Could I request that `aeson` released have their upper version bounds on `base` revised to `< 4.13`? The lack of upper version bounds is currently causing failing build plans to be chosen with GHC 8.8.1 on Travis. See [here](https://travis-ci.org/lens/lens-aeson/jobs/525618676#L814) for one example.

**Comment by phadej** [comment]
haskell-ci script have allow-newer: base, so how upper bounds would help?

**Comment by RyanGlScott** [comment]
I edited my previous comment to state that realization, although I now realize that you wouldn't see that if you're only reading e-mail notifications. My bad.

**Comment by phadej** [comment]
No worries

It’s good to know that there will be breakage, this issue is exactly to collect the list; and then fix stuff

**Comment by quasicomputational** [comment]
Two more for the pile needing revisions of old versions:

* `colour`

* `json`

**Comment by cartazio** [comment]
curiously: in some CI i've got, i'm seeing the bad time version picked up for 8.8.1RC / hackage head https://travis-ci.org/haskell/primitive/jobs/533885678#L912

**Comment by recursion-ninja** [comment]
I think we may need to add `polyparse` to the list since the [hackage "bug tracker"]([url](http://hackage.haskell.org/package/polyparse)) link directs here and `polyparse` doesn't build with the alpha version of `ghc-8.8.1`.

**Comment by phadej** [comment]
@recursion-ninja that's different, `polyparse` has the correct bounds, but indeed might not build with GHC-8.8.1 if you relax them. That's different issue. And seems as I uploaded polyparse, I might actually do another upload when it's time. 

There's a bunch of packages in https://github.com/hackage-trustees/malcolm-wallace-universe which I completely forgot about.

**Comment by recursion-ninja** [comment]
@phadej I'm not sure what you mean. `polyparse` still looks broken to me. If there's an updated repository from which I can get newer, compatible code I would love to set that in my `project.cabal` and/or `stack.yaml` file(s).

```
/tmp/stack31990/polyparse-1.12.1/src/Text/ParserCombinators/HuttonMeijer.hs:66:4: error:
polyparse                        >     ‘fail’ is not a (visible) method of class ‘Monad’
polyparse                        >    |
polyparse                        > 66 |    fail            = Fail.fail
polyparse                        >    |    ^^^^
polyparse                        >    

```

**Comment by quasicomputational** [comment]
`memory` will also need its bounds adjusted.

**Comment by phadej** [comment]
The maintainer author of `memory` have been asked multiple times to put version bounds on dependencies, in `memory` and other packages.

It's one reason, why I simply avoid packages by that person. (It's difficult, but not impossible).

Please, open issues to their issue trackers, as that author simply ignores anything coming from Hackage trustees.

TL;DR I personally refuse to revise `memory` as the author is ignorant for the work trustees do.

**Comment by phadej** [comment]
@vmchale `time` is "fine". it works ok when it's a dependency.

**Comment by hvr** [comment]
- `vector-th-unbox revised` (upstream ticket https://github.com/liyang/vector-th-unbox/issues/22#issuecomment-527014591)

- `json` revised (cc @yav)
    - https://hackage.haskell.org/package/json-0.9.3/revisions/
    - https://hackage.haskell.org/package/json-0.9.2/revisions/

**Comment by cartazio** [comment]
@phadej to clarify/expand: you mean that the build failure on hackage matrix happens only when its a local source tree / checkout build, rather than normally as a pkg we cabal install, right?

**Comment by vmchale** [comment]
With the latest `happy` release, I believe the `language-c` bounds are now going to cause breakages.

**Comment by phadej** [comment]
`language-c` reviewed.

**Comment by athas** [comment]
`vector-binary-instances` could also do with a bump for `base`.  There is a [upstream pull request](https://github.com/bos/vector-binary-instances/pull/18).

**Comment by phadej** [comment]
not allowing newer base is **not a breakage**, as far as this issue is concerned.

**Comment by quasicomputational** [comment]
`hnix` will also be broken soon, but that's being masked by dependency failures.

**Comment by phadej** [comment]
@quasicomputational hnix is in miserable state even without masking: https://matrix.hackage.haskell.org/#/package/hnix

**Comment by phadej** [comment]
I made revisions for ~4 recent years of `jose` versions. That should be good enough.
